### PR TITLE
Filter move

### PR DESCRIPTION
**Pull request by cityiron**
**What this PR does**:
Filter move to self package and create responseFilter

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by yqxu — APPROVED** [review]

File: pkg/common/constant/proxy.go [hunk 1]
```diff
@@ -19,13 +19,27 @@ package constant
...
 
-// body passthrough
+// body passthrough.
 const (
 	DefaultBodyAll = "_all"
 )
+
+// proxy code.
+const (
+	TimeoutError    = "DGP1005"
```
Comment (williamfeng323) [inline pkg/common/constant/proxy.go:35]: Do we need the const error message or we can use 408 status code?

File: pkg/filter/response/response.go [hunk 2]
```diff
@@ -15,22 +15,75 @@
...
+	contexthttp "github.com/dubbogo/dubbo-go-proxy/pkg/context/http"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/filter"
 )
 
-// NewDubboResponse create dubbo response
-func NewDubboResponse(data interface{}) *client.Response {
-	r, err := dealResp(data, true)
+func init() {
+	strategy := os.Getenv(constant.EnvResponseStrategy)
+	if strategy == "" {
```
Comment (williamfeng323) [inline pkg/filter/response/response.go:40]: use len(strategy) != 0?

File: pkg/common/constant/proxy.go (hunk 1 above)
Comment (cityiron) [reply pkg/common/constant/proxy.go]: agree

File: pkg/filter/response/response.go (hunk 2 above)
Comment (cityiron) [reply pkg/filter/response/response.go]: yes, better

**Review by williamfeng323 — APPROVED** [review]

**Review by zouyx — APPROVED** [review]

**Pull request merged**